foundationdb/src/hca.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2018 foundationdb-rs developers, https://github.com/bluejekyll/foundationdb-rs/graphs/contributors
+// Copyright 2013-2018 Apple, Inc and the FoundationDB project authors.
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! Docs!
+//!
+
+use std::sync::Mutex;
+
+use byteorder::ByteOrder;
+use futures::future::Future;
+use futures::stream::Stream;
+use rand::Rng;
+
+use error::Error;
+use options::{ConflictRangeType, MutationType, TransactionOption};
+use subspace::Subspace;
+use transaction::{RangeOptionBuilder, Transaction};
+use tuple::Element;
+
+lazy_static! {
+  static ref LOCK: Mutex<i32> = Mutex::new(0);
```

Comment:
Is this lock truly global? or should it be limited to each instance of an HCA?

Reply:
still not sure about this one. I'll dig into the other bindings some more